packages/@aws-cdk/aws-cognito/lib/user-pool-client.ts

```diff
@@ -343,12 +343,20 @@ export class UserPoolClient extends Resource implements IUserPoolClient {
   }
 
   private configureAuthFlows(props: UserPoolClientProps): string[] | undefined {
+    if (!props.authFlows) return undefined;
+
     const authFlows: string[] = [];
-    if (props.authFlows?.userPassword) { authFlows.push('ALLOW_USER_PASSWORD_AUTH'); }
-    if (props.authFlows?.adminUserPassword) { authFlows.push('ALLOW_ADMIN_USER_PASSWORD_AUTH'); }
-    if (props.authFlows?.custom) { authFlows.push('ALLOW_CUSTOM_AUTH'); }
-    if (props.authFlows?.userSrp) { authFlows.push('ALLOW_USER_SRP_AUTH'); }
-    if (props.authFlows?.refreshToken) { authFlows.push('ALLOW_REFRESH_TOKEN_AUTH'); }
+    if (props.authFlows.userPassword) { authFlows.push('ALLOW_USER_PASSWORD_AUTH'); }
+    if (props.authFlows.adminUserPassword) { authFlows.push('ALLOW_ADMIN_USER_PASSWORD_AUTH'); }
+    if (props.authFlows.custom) { authFlows.push('ALLOW_CUSTOM_AUTH'); }
+    if (props.authFlows.userSrp) { authFlows.push('ALLOW_USER_SRP_AUTH'); }
+
+    // refreshToken should always be allowed if authFlows are present
+    if (props.authFlows.refreshToken === undefined) {
+      authFlows.length && authFlows.push('ALLOW_REFRESH_TOKEN_AUTH');
+    } else {
+      props.authFlows.refreshToken && authFlows.push('ALLOW_REFRESH_TOKEN_AUTH');
+    }
```

Comment:
seems like we could clean this up a little bit - it shouldn't be undefined since we're defining it on line 348 and initializing it to be empty.

```suggestion
    if (authFlows.length > 0 || props.authFlows?.refreshToken) {
      authFlows.push('ALLOW_REFRESH_TOKEN_AUTH')
    }
```

Reply:
Will do.